src/codecache/CodeCache.cpp

```diff
@@ -0,0 +1,399 @@
+/*
+ * Copyright (c) 2020-present Samsung Electronics Co., Ltd
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
+ *  USA
+ */
+
+
+#if defined(ENABLE_CODE_CACHE)
+
+#include "Escargot.h"
+#include "codecache/CodeCache.h"
+#include "codecache/CodeCacheReaderWriter.h"
+#include "parser/Script.h"
+#include "parser/CodeBlock.h"
+
+namespace Escargot {
+
+CodeCache::CodeCache()
+    : m_writer(new CodeCacheWriter())
+    , m_reader(new CodeCacheReader())
+    , m_stringTableMap(new CacheStringTableMap())
+{
+}
+
+CodeCache::~CodeCache()
+{
+    delete m_writer;
+    delete m_reader;
+    delete m_stringTableMap;
+}
+
+CacheStringTable* CodeCache::addStringTable(Script* script)
+{
+    ASSERT(m_stringTableMap->find(script) == m_stringTableMap->end());
+
+    CacheStringTable* table = new CacheStringTable();
+    m_stringTableMap->insert(std::make_pair(script, table));
+    return table;
+}
+
+void CodeCache::loadMetaInfos(Script* script)
+{
+    char metaFileName[128] = { 0 };
```

Comment:
IMO 128 bytes is not enough space.
Script src can be very long(URL, filepath)
I think we need at least PATH_MAX

Reply:
Thanks for the review.
This patch is the first commit of CodeCache and there are a lot of things should be enhanced.
File name is one of them. Now the source file name is temporally used for debugging purpose.
But I'm considering the hash value of the file name string because file name has variant string length depending on the situation.
I'll update it in the following patches.